velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -264,6 +266,136 @@ TpchPlan TpchQueryBuilder::getQ3Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ5Plan() const {
+  std::vector<std::string> customerColumns = {"c_custkey", "c_nationkey"};
+  std::vector<std::string> ordersColumns = {
+      "o_orderdate", "o_custkey", "o_orderkey"};
+  std::vector<std::string> lineitemColumns = {
+      "l_suppkey", "l_discount", "l_extendedprice", "l_orderkey"};
+  std::vector<std::string> supplierColumns = {"s_nationkey", "s_suppkey"};
+  std::vector<std::string> nationColumns = {
+      "n_nationkey", "n_name", "n_regionkey"};
+  std::vector<std::string> regionColumns = {"r_regionkey", "r_name"};
+
+  auto customerSelectedRowType = getRowType(kCustomer, customerColumns);
+  const auto& customerFileColumns = getFileColumnNames(kCustomer);
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto supplierSelectedRowType = getRowType(kSupplier, supplierColumns);
+  const auto& supplierFileColumns = getFileColumnNames(kSupplier);
+  auto nationSelectedRowType = getRowType(kNation, nationColumns);
+  const auto& nationFileColumns = getFileColumnNames(kNation);
+  auto regionSelectedRowType = getRowType(kRegion, regionColumns);
+  const auto& regionFileColumns = getFileColumnNames(kRegion);
+
+  std::string regionNameFilter = "r_name = 'ASIA'";
+  const auto orderDate = "o_orderdate";
+  std::string orderDateFilter = formatDateFilter(
+      orderDate, ordersSelectedRowType, "'1994-01-01'", "'1994-12-31'");
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId customerScanNodeId;
+  core::PlanNodeId ordersScanNodeId;
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId supplierScanNodeId;
+  core::PlanNodeId nationScanNodeId;
+  core::PlanNodeId regionScanNodeId;
+
+  auto region = PlanBuilder(planNodeIdGenerator)
+                    .tableScan(
+                        kRegion,
+                        regionSelectedRowType,
+                        regionFileColumns,
+                        {regionNameFilter})
+                    .capturePlanNodeId(regionScanNodeId)
+                    .planNode();
+
+  auto nationRegionJoinNode =
```

Comment:
`nationJoinRegion` probably helps since we can then apply the same to others without ambiguity?